### ISSUE TITLE
IntelliJ EAP offers to install ancient 0.1.2 version of the plugin

### ISSUE THREAD

**Issue by intgr** [koxudaxi/poetry-pycharm-plugin#233]
**Describe the bug**
I'm using IntelliJ IDEA 2021.2 EAP (Build #IU-212.3116.43). If I search for the Poetry plugin in the plugin manager, it offers to install the ancient version 0.1.2.

<img width="1012" alt="image" src="https://user-images.githubusercontent.com/137616/120172825-1dc86300-c20c-11eb-8e94-b4a8da0e64d2.png">

However, if I look on the web https://plugins.jetbrains.com/plugin/14307-poetry , the newest plugin version appears 1.1.4

If I try to enable this plugin, it fails with the error: java.lang.NoSuchMethodError: 'void com.jetbrains.python.packaging.PyPackageUtil.runOnChangeUnderInterpreterPaths(com.intellij.openapi.projectRoots.Sdk, java.lang.Runnable)'

<details>

```
java.lang.NoSuchMethodError: 'void com.jetbrains.python.packaging.PyPackageUtil.runOnChangeUnderInterpreterPaths(com.intellij.openapi.projectRoots.Sdk, java.lang.Runnable)'
	at com.koxudaxi.poetry.PyPoetryPackageManager.<init>(PyPoetryPackageManager.kt:34)
	at com.koxudaxi.poetry.PyPoetryPackageManagersImpl.forSdk(PyPoetryPackageManagersImpl.kt:19)
	at com.koxudaxi.poetry.PyPoetryPackageManager$Companion.getInstance(PyPoetryPackageManager.kt:151)
	at com.koxudaxi.poetry.PoetryPackageRequirementsInspection$Visitor.findUnsatisfiedRequirements(PoetryPackageRequirementsInspection.kt:117)
	at com.koxudaxi.poetry.PoetryPackageRequirementsInspection$Visitor.checkPackagesHaveBeenInstalled(PoetryPackageRequirementsInspection.kt:138)
	at com.koxudaxi.poetry.PoetryPackageRequirementsInspection$Visitor.visitPyFile(PoetryPackageRequirementsInspection.kt:84)
	at com.jetbrains.python.psi.impl.PyFileImpl.accept(PyFileImpl.java:261)
	at com.intellij.codeInspection.InspectionEngine.acceptElements(InspectionEngine.java:64)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.lambda$visitRestElementsAndCleanup$7(LocalInspectionsPass.java:352)
	at com.intellij.util.AstLoadingFilter.lambda$toComputable$2(AstLoadingFilter.java:174)
	at com.intellij.util.AstLoadingFilter.disallowTreeLoading(AstLoadingFilter.java:132)
	at com.intellij.util.AstLoadingFilter.disallowTreeLoading(AstLoadingFilter.java:121)
	at com.intellij.util.AstLoadingFilter.disallowTreeLoading(AstLoadingFilter.java:111)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.lambda$visitRestElementsAndCleanup$10(LocalInspectionsPass.java:352)
	at com.intellij.concurrency.ApplierCompleter.execAndForkSubTasks(ApplierCompleter.java:136)
	at com.intellij.concurrency.ApplierCompleter.execAndForkSubTasks(ApplierCompleter.java:149)
	at com.intellij.concurrency.ApplierCompleter.execAndForkSubTasks(ApplierCompleter.java:149)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1108)
	at com.intellij.concurrency.ApplierCompleter.lambda$wrapInReadActionAndIndicator$1(ApplierCompleter.java:92)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:705)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:647)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:63)
	at com.intellij.concurrency.ApplierCompleter.wrapInReadActionAndIndicator(ApplierCompleter.java:104)
	at com.intellij.concurrency.ApplierCompleter.lambda$compute$0(ApplierCompleter.java:83)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:174)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:223)
	at com.intellij.concurrency.ApplierCompleter.compute(ApplierCompleter.java:83)
	at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:746)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```
</details>

**Environments (please complete the following information):**
 - IDE: IntelliJ IDEA 2021.2 EAP (Build #IU-212.3116.43)
 - OS: macOS
 - Plugin version 0.1.2

**Comment by koxudaxi** [comment]
@intgr 
Thank you for creating this issue.

> I'm using IntelliJ IDEA 2021.2 EAP (Build #IU-212.3116.43). If I search for the Poetry plugin in the plugin manager, it offers to install the ancient version 0.1.2.

OK, I will update the plugin for 2021.2 EAP
I always meet the problem when a new EAP comes up :cry: 
I want to get new EAP information automatically :thinking:

**Comment by intgr** [comment]
I wonder if there's any way to revoke old versions? So that IDE would say "this extension is not compatible", instead of installing the old version.

**Comment by koxudaxi** [comment]
I can set a supporting version range. But the old version is published without the limit.

There is a version list here.
https://plugins.jetbrains.com/plugin/14307-poetry/versions

I should delete the old version what is unset the version limit.

**Comment by koxudaxi** [comment]
> I should delete the old version what is unset the version limit.

I have hidden the old versions.
I will create a new version supporting  2021.2 EAP

**Comment by koxudaxi** [comment]
@intgr 
I have released new version `1.1.4-212` for 2021.2 EAP :tada:

**Comment by intgr** [comment]
Thanks :+1:
